### DISCUSSION TITLE
Automation based on User Table (Row Created)

### DISCUSSION THREAD

**Discussion by SantoshSrinivas79** [Budibase/budibase#4580] [Help]
Hey guys,

Trying to figure this automation out

I want to create a row in a table called `User Profiles` when a User is created.
I have a column called `User ID` in `User Profiles` which should contain the `_id` of the user that was created in the app.

Please advise.

**Comment by mjashanks** [comment]
Hey @SantoshSrinivas79 - I have tried this and cannot get it to work either.

It sort of doesn't make a lot of sense to have a "Row Created" trigger on the users table, as users are created globally. 

However, when a user gets assigned to an app they do get added to the users table, within the app. I would have expected the trigger to fire in this scenario - which it does not.

It's worth noting that the trigger will only fire on a published app, whereas all the data in the `Data` section belongs to the unpublished app. I had to create a view of my `User Profile` table inside the published app to test this. It didn't matter though - doesn't look like the trigger is firing for users.

Out of interest, what are you trying to do? There are 2 interpretation of `Row Created on User Table`
1. Fire the automation when a user is created (Globally)
2. Fire the automation when an existing user is given access to the app.

**Reply by SantoshSrinivas79** [reply]
Hi @mjashanks .. mainly this one

> Fire the automation when an existing user is given access to the app.

Basically, I don't want to have write permissions on the `Users` table .... I am sort of OK that users can modify `User Profiles`.
Ideally, users should be able to modify only their data but I believe that is not possible (i.e. everyone at same role can modify each other)
